CMakeLists.txt

```diff
@@ -13,10 +13,6 @@ set(BUILD_ARCH native CACHE STRING "Compile for this CPU architecture.")
 # Custom CMake options
 option(COMPILE_CPU "Compile CPU version" ON)
 option(COMPILE_CUDA "Compile GPU version" ON)
-option(COMPILE_CUDA_SM35 "Compile GPU version with SM35 support" ON)
-option(COMPILE_CUDA_SM50 "Compile GPU version with SM50 support" ON)
-option(COMPILE_CUDA_SM60 "Compile GPU version with SM60 support" ON)
-option(COMPILE_CUDA_SM70 "Compile GPU version with SM70 support" ON)
```

Comment:
I considered this and opted against it because this solution seemed hacky for me: https://stackoverflow.com/questions/3766740/overriding-a-default-option-value-in-cmake-from-a-parent-cmakelists-txt
eg, what happens if we disable the option but the user specifically enables it? Then the compilation might fail due to unsupported compute architecture?  It also wasn't clear for me what takes precedence? The `CMakeLists.txt` option or the user command line input when invoking `cmake` but I can test that easily. Still, I don't think it's a better solution.